AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -3501,6 +3474,27 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return [self layoutThatFits:constrainedSize parentSize:constrainedSize.max];
 }
 
+- (void)setPreferredFrameSize:(CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // Deprecated preferredFrameSize just calls through to set width and height
+  _style.width = ASDimensionMake(preferredFrameSize.width);
+  _style.height = ASDimensionMake(preferredFrameSize.height);
+  [self invalidateCalculatedLayout];
+}
+
+- (CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  if (_style.width.unit == ASDimensionUnitPoints && _style.height.unit == ASDimensionUnitPoints) {
+    return CGSizeMake(_style.width.value, _style.height.value);
+  }
+
+  return CGSizeZero;
```

Comment:
Should this assert? Something along the lines of "Could not figure out the preferredFrameSize, it's recommended that you switch to accessing height and width directly."